### DISCUSSION TITLE
Custom Theme is not SSR? App styles flicker on page load.

### DISCUSSION THREAD

**Discussion by eleazareramos** [heroui-inc/heroui#286] [Help] [answered]
Hi there, after following the [documentation](https://nextui.org/docs/theme/customize-theme) on creating a custom theme, it seems like the theme is not injected into the document server side. It is served with the default theme, and then when loaded on the client, it converts to the new theme.

See in this video how it starts with the default primary blue for a quick second, and then the custom theme kicks in changing it to the desired brand color.
https://user-images.githubusercontent.com/25910203/154084143-c1af33d2-6b0f-4b8d-b0d9-ef6993a2b2fd.mov

Here is my code:

pages/_app.js
```
import theme from '../styles/theme'
import { NextUIProvider } from '@nextui-org/react'

function App({ Component, pageProps }) {
  return (
    <NextUIProvider theme={theme}>
      <Component {...pageProps} />
    </NextUIProvider>
  )
}

export default App
```

pages/_document.js
```
import Document, { Html, Head, Main, NextScript } from 'next/document'
import { CssBaseline } from '@nextui-org/react'

class AppDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx)
    return {
      ...initialProps,
      styles: <>{initialProps.styles}</>,
    }
  }

  render() {
    return (
      <Html lang="en">
        <Head>{CssBaseline.flush()}</Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default AppDocument
```

styles/theme.js
```
import { createTheme } from '@nextui-org/react'

export default createTheme({
  type: 'light',
  theme: {
    colors: {
      primary: '#3f5765'
    }
  },
})
```

In other UI frameworks, such as styled-components, there is a provided method to produce a `ServerStyleSheet` and load it into `pages/_document.js` ([docs here](https://styled-components.com/docs/advanced#server-side-rendering)) but it seems like that concept isn't present with NextUI.

Am I missing something?

**Comment by jrgarciadev** [comment, marked as answer]
Hey @eleazareramos I was looking into this subject and you are right NextUI has a flash theme switch when you set custom values, in the meanwhile, you can use [next-themes](https://github.com/pacocoursey/next-themes) for fixing it as follows:

```jsx
// _app.js

import { createTheme, NextUIProvider } from "@nextui-org/react";
import { ThemeProvider as NextThemesProvider } from "next-themes";

const customTheme = createTheme({
    type: "dark",
    theme: {
        colors: {
	   primary: "#AAFFEC",
        },
    },
});

function App({ Component, pageProps }) {
  return (
    <NextThemesProvider
	  defaultTheme="system"
	  attribute="class"
	  value={{
	    dark: customTheme.className,
	  }}
    >
	<NextUIProvider>
           <Component {...pageProps} />
	</NextUIProvider>
    </NextThemesProvider>
  );
}

export default App;
```

We will try to fix this as soon as possible 👍🏻

**Reply by eleazareramos** [reply]
Great. Thank you!!!

**Reply by devbravo** [reply]
Got this error after adding your solution. It did solve the flickering problem: Do not add <script> tags using next/head (see inline <script>). Use next/script instead.

**Reply by antoinelin** [reply]
Hi, actually this fix is not working for me, the default colors and styles keeps flashing at the first render before changes to the custom ones, any idea?

**Comment by devbravo** [comment]
Great. It worked! Question I'm in a team that's building a startup and decided to use nextui for ui. We could've chosen material ui or bootstrap, but decided to go with nextui because it looks good and it's backed by Vercel I believe, although it's still in beta, we still decided to give it a try. As a startup we dont want to much overhead regarding the technologies we use. So my question is how long will it nextui stay in beta?? When will there be a stable release

**Reply by zvictor** [reply]
Nextui is not backed by Vercel.

**Comment by jansedlon** [comment]
That is awesome, this solution even works for Remix.run!

**Comment by tjwds** [comment]
Is there a matching issue for this question?  I also ran into this problem, and I was helped by a contributor on the Discord who pointed me here, but I'd like to not need the additional `next-themes` dependency. :-)

**Comment by leogilmour** [comment]
This hasn't fully fixed the flash for me. (it is a less noticeable, but still there)

package.json:
```
"next-themes": "^0.2.1",
"@nextui-org/react": "1.0.0-beta.10",
```
_app.tsx
```
import type { ReactElement, ReactNode } from "react";
import type { NextPage } from "next";
import type { AppProps } from "next/app";
import { NextUIProvider } from "@nextui-org/react";
import myTheme from "../styles/defaultTheme";
import { ThemeProvider as NextThemesProvider } from "next-themes";

export type NextPageWithLayout<P = Record<string, never>, IP = P> = NextPage<
  P,
  IP
> & {
  getLayout?: (page: ReactElement) => ReactNode;
};

type AppPropsWithLayout = AppProps & {
  Component: NextPageWithLayout;
};

function MyApp({ Component, pageProps }: AppPropsWithLayout) {
  // Use the layout defined at the page level, if available
  const getLayout = Component.getLayout ?? ((page) => page);

  return (
    <NextThemesProvider
      defaultTheme="light"
      attribute="class"
      enableSystem={false}
      themes={["light"]}
      value={{
        light: myTheme.className,
      }}
    >
      <NextUIProvider>{getLayout(<Component {...pageProps} />)}</NextUIProvider>
    </NextThemesProvider>
  );
}

export default MyApp;
```
defaultTheme.tsx
```
import { createTheme } from "@nextui-org/react";

const myTheme = createTheme({
  type: "light", 
  theme: {
    colors: {
      primary: "#4D46FF", // purple
      ...
  }}
});

export default myTheme;
```

Any help would be GREAT -- maybe I've missed something / implemented next-themes incorrectly
